demos/segment-anything/index.js

```diff
@@ -626,18 +628,21 @@ const ui = async () => {
     await setupORT("segment-anything", "dev");
     showCompatibleChromiumVersion("segment-anything");
 
-    if (config.devicetype.toLowerCase().indexOf("cpu") > -1 || config.provider.toLowerCase().indexOf("wasm") > -1) {
+    const deviceType = config.deviceType.toLowerCase();
+    const provider = config.provider.toLowerCase();
+
+    if (deviceType.indexOf("cpu") > -1 || provider.indexOf("wasm") > -1) {
         device.innerHTML = "CPU";
         badge.setAttribute("class", "cpu");
         document.body.setAttribute("class", "cpu");
     } else if (
-        config.devicetype.toLowerCase().indexOf("gpu") > -1 ||
-        config.provider.toLowerCase().indexOf("webgpu") > -1
+        deviceType.indexOf("gpu") > -1 ||
+        provider.indexOf("webgpu") > -1
```

Comment:
Is there any risk using `indexOf` here instead of a straight comparison? `provider == "webgpu"`

I worry that (maybe someone else later, if not you) someone could write logic ordered like this...

```
if (provider.indexOf("gpu") > -1) {
    ...
} else if (provider.indexOf("webgpu") > -1) {
    ...
}
```

...and the first `if` statement would silently consume the "webgpu" case because "gpu" is within "webgpu".

Reply:
Agree, thanks for the great feedback.

Reply:
(of course, my specific example above is silly (I grant that), because there is no "provider" of plain "gpu", but the concept applies nonetheless 😉)